Task1 changed

mustafakamber opened the pull request.

mustafakamber committed:
task1 project file added

mustafakamber committed:
project restored

mustafakamber committed:
Güncellenmiş kodlar eklendi

mustafakamber committed:
Proje güncellendi

nezih94 commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,26 @@
+    <dimen name="largeTextSize">20sp</dimen>
+    <dimen name="hugeTextSize">64sp</dimen>
+
+    <!-- MarginSize -->
```
İsimlendirme konusunda bence doğru yoldayız. Margin dimen değerlerini isimlendirirken, kullanım alanını belirten daha specific context verebiliriz. 

Ör: "normalMarginSize" yerine "regularHorizontalSpacing", "regularVerticalSpacing"

mustafakamber committed:
dimens.xml updated

mustafakamber committed:
task1 updated

mustafakamber committed:
dimens.xml updated

iamkurtgoz commented on task-1-solution/FirstTaskSolution/app/src/main/res/values-night/themes.xml:
```diff
@@ -0,0 +1,16 @@
+<resources xmlns:tools="http://schemas.android.com/tools">
+    <!-- Base application theme. -->
+    <style name="Theme.FirstTaskSolution" parent="Theme.MaterialComponents.DayNight.NoActionBar">
```
Burada default attr değerleri kalmış gibi görünüyor uygulama gündüz modunda gece modu gibi görünüyor ve gece modunda renkler bozuluyor. Eğer gece gündüz modu kullanacaksan buradaki renkleri düzenlemen gerekir eğer kullanmayacaksan gündüz modundaki değerleri aynı şekilde buraya kopyala-yapıştır.

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,84 @@
+    <dimen name="large600VerticalSpacing">112dp</dimen>
+    <dimen name="large700VerticalSpacing">116dp</dimen>
+    <dimen name="large800VerticalSpacing">120dp</dimen>
+    <dimen name="large900VerticalSpacing">124dp</dimen>
```
Eline sağlık düzenli olmuş ama layout içerisinde bu büyüklüklerde spacing kullanmak uygun değil. Küçük boyutlu ekranlarda bozulmalara sebebiyet verebilir.

En büyük spacing değeri `32dp` olabilir. İstisnai durumlarda `48dp`'ye çıkabilir.

Aralarda ekstradan boşluk vermeden `constraint`'leri ayarlayabilmek `ConstraintLayout`'un bize sağladığı kolaylıklardan zaten. Bunu kullanmazsak, `ConstraintLayout`'dan tam faydalanmamış oluruz.

O yüzden bu task kapsamında `48dp` üzerindekileri kullanmayalım. Bundan sonraki task'larda da mümkün mertebe daha da düşük boşluklar kullanalım.

Konuştuğumuz sadece spacingler için. `width` ve `height` değerleri için bu kural daha fazla esnetilebilir.

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,84 @@
+    <dimen name="large700HorizontalSpacing">116dp</dimen>
+    <dimen name="large800HorizontalSpacing">120dp</dimen>
+    <dimen name="large900HorizontalSpacing">124dp</dimen>
+
```
Gereksiz boşlukları kaldıralım. Tek boşluk olabilir.

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,84 @@
+    <dimen name="mediumImageSize">24dp</dimen>
+    <dimen name="largeImageSize">64dp</dimen>
+
+
```
2 satır boşluk , 1'e düşebilir.

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/strings.xml:
```diff
@@ -0,0 +1,19 @@
+    <string name="nextDayGroupFourthDateText">Cum, 05 Nisan</string>
+    <string name="nextDayGroupTemperatureRangesText">14° - 27°</string>
+
+
```
Gereksiz boşluk

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,87 @@
+        <item name="android:textColor">?android:attr/textColorPrimary</item>
+    </style>
+
+
```
Gereksiz boşluk

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,87 @@
+        <item name="android:textColor">?android:attr/textColorPrimary</item>
+    </style>
+
+
```
Gereksiz boşluk

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/themes.xml:
```diff
@@ -0,0 +1,87 @@
+        <item name="android:layout_height">@dimen/largeImageSize</item>
+    </style>
+
+
```
Gereksiz boşluk

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,84 @@
+    <dimen name="largeTextSize">20sp</dimen>
+    <dimen name="hugeTextSize">64sp</dimen>
+
+    <!-- VerticalSpacing -->
```
Bu yorum boşta kalmış. Kategorize etmek içinse, gerek yok çünkü altında ilgili kod bloğu yok. Kaldırabiliriz.

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/drawable/ic_launcher_background.xml:
```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="108dp"
```
Hardcoded size

akndmr commented on task-1-solution/FirstTaskSolution/app/src/main/res/drawable-v24/ic_launcher_foreground.xml:
```diff
@@ -0,0 +1,30 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:width="108dp"
```
Hardcoded size

akndmr reviewed (changes requested):
Eline sağlık Mustafa, genel olarak düzenli ama bazı kısımlarda düzeltilmesi gereken noktalar var.
Özellikle spacing'lere bakış açımızı değiştirmemiz gerekiyor. Büyük spacing'ler olmadan `constraintlayout'`un nimetlerinden yararlanarak, layout'u tekrar düzenlemek gerekli.

mustafakamber committed:
dimens.xml,themes.xml,strings.xml updated

nezih94 commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,22 @@
+    <dimen name="smallTextSize">14sp</dimen>
+    <dimen name="mediumTextSize">16sp</dimen>
+    <dimen name="largeTextSize">20sp</dimen>
+    <dimen name="wideTextSize">64sp</dimen>
```
"large'dan daha büyük" anlamına gelecek kelime bence "Wide" olmamalı. Daha uygun bir İngilizce kelime kullanabiliriz, örneğin "Huge"

mustafakamber replied:
Düzenledim hocam

mustafakamber committed:
converted from wide to huge

mustafakamber committed:
themes.xml spaces deleted

fatih-ozturk commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,238 @@
+        android:text="@string/degreeMarkText"
+        app:layout_constraintBottom_toBottomOf="@+id/temperatureText"
+        app:layout_constraintStart_toEndOf="@+id/temperatureText"
+        app:layout_constraintTop_toTopOf="@+id/temperatureText" />
```
referans verirken `@+id/` yerine `@id/` kullanalim.

https://medium.com/@nikhilschumifanikhil/android-basics-difference-between-id-and-id-f4eb66915361


```suggestion
        app:layout_constraintBottom_toBottomOf="@id/temperatureText"
        app:layout_constraintStart_toEndOf="@id/temperatureText"
        app:layout_constraintTop_toTopOf="@id/temperatureText" />
```

fatih-ozturk commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,238 @@
+            android:id="@+id/nextFirstDayDateText"
+            style="@style/smallRegularText"
+            android:layout_alignParentStart="true"
+            android:layout_centerVertical="true"
```
bunlar relative layout'a ozel constraint layoutta islevsiz kaliyor. bu tarz problemleri duzeltelim.

scotoglu commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,238 @@
+            android:layout_alignParentStart="true"
+            android:layout_centerVertical="true"
+            android:layout_marginTop="@dimen/mediumVerticalSpacing"
+            android:layout_marginBottom="17dp"
```
dimens.xml'den alabilir misin

scotoglu commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,238 @@
+        <TextView
+            android:id="@+id/nextSecondDayDateText"
+            style="@style/smallRegularText"
+            android:layout_alignParentStart="true"
```
alignParentEnd, alignParentStart, centerVertical attributes relative layout ile birlikte kullanılıyor. Constraint layout içinde herhangi bir etkisi bulunmuyor. Bunları kaldırabilir misin. [Detay](https://halilozel1903.medium.com/android-relative-layout-kullanımı-394c75902d9d)

mustafakamber committed:
feat!: I used @id/ instead of @+id/ when referencing
feath!: alignParentEnd, alignParentStart, centerVertical deleted from Views
feath!:android:layout_marginBottom="17dp" read from dimens.xml

mustafakamber committed:
feat!: statusBarColor changed
feath!: hugeVerticalSpacing deleted in dimens.xml

scotoglu commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,224 @@
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/nextSecondDayDateText"
```
tasarımda tarih gösterimi Çar, 03 Nisan formatında, peki bu tarihi Çarşamba, 03 Nisan 2024 şeklinde göstermek istesek nasıl olurdu? Metni değiştirdikten sonra, activity_weather.xml > design kısmında, sağ alt köşedeki ikon yardımıyla farklı ekranlar için kontrol edebilir misin?
![Screenshot 2024-03-05 at 20 31 51](https://github.com/mustafakamber/mobillium-intern2024-task-solution/assets/38690766/8689e957-4c6a-4894-82b0-e15ade36ca79)

nezih94 replied:
@scotoglu Sefa ben bunu bilmiyordum :D

scotoglu commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -0,0 +1,224 @@
+        android:id="@+id/cityText"
+        style="@style/largeMediumText"
+        android:layout_marginTop="@dimen/largeVerticalSpacing"
+        android:text="@string/cityNameText"
```
İstanbul yerine, [buranın](https://tr.wikipedia.org/wiki/Taumatawhakatangihangakoauauotamateaturipukakapikimaungahoronukupokaiwhenuakitanatahu#:~:text=Diğer%20ismi%20olan%20%22Taumatawhakatangihangakoauauotamateaurehaeaturipukakapikimaungahoronukupokaiwhenuakitanatahu%22%2092,ise%20105%20harften%20oluşmaktadır%2C%20bu) hava durumunu gösterdiğimizde,tasarımda bazı bozulmalar oluyor. Yine İstanbul metni kalsın ama diğer yerde gelicekmiş gibi genişlik ve kenar boşluklarını düzenleyebilir misin. Start ve End 16dp verilebilir yine.

scotoglu reviewed (commented):
Daha uzun textller geldiğinde TextView'lerin nasıl davrandığını görmek için `tools:text="@tools:sample/lorem[WordCount]"` kullanabilirsin.

akndmr reviewed (approved):
Sefa'nın da yorumları var sanırım. Onun dışında tamamdır. Eline sağlık 👏

mustafakamber committed:
feat!: @id/cityText width = match_parent, margin(Start+End)=16dp,textAlignment = center
feath!: hugeVerticalSpacing added  in dimens.xml

mustafakamber committed:
Update strings.xml

mustafakamber committed:
feat!: night\themes.xml updated based on themes.xml

mustafakamber committed:
Merge remote-tracking branch 'origin/task1' into task1

mustafakamber committed:
feat!: constraintLayout background read from colorSecondary in themes.xml

mustafakamber committed:
feat!: ...dateText updated from strings.xml

mustafakamber committed:
feat!: ...dateText updated from strings.xml

nezih94 commented on task-1-solution/FirstTaskSolution/app/src/main/res/layout/activity_weather.xml:
```diff
@@ -7,7 +7,7 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="?android:attr/textColorSecondary"
+    android:background="?attr/colorSecondary"
```
Root layout background rengi için colorSecondary attribute kullanımı Material renk kurallarına göre çoğunlukla uygun değil. Mesela bunun yerine surface renk attribute'u kullanılabilir.

Material 2 ve Material 3 color system dökümantasyonlarına göz atabilirsin.
https://m2.material.io/design/color/the-color-system.html

mustafakamber committed:
feat!: constraintLayout background read from colorSurface in themes.xml

mustafakamber committed:
feat!: themes.xml and night\themes.xml color updated

iamkurtgoz commented on task-1-solution/FirstTaskSolution/app/src/main/res/values/themes.xml:
```diff
@@ -1,20 +1,21 @@
         <!-- Customize your theme here. -->
+        <item name="colorSurface">@color/dark</item>
     </style>
     <style name="smallRegularText">
```
bir satır boşluk bırakalım

iamkurtgoz commented on task-1-solution/FirstTaskSolution/app/src/main/res/values-night/themes.xml:
```diff
@@ -1,20 +1,21 @@
         <!-- Customize your theme here. -->
+        <item name="colorSurface">@color/dark</item>
     </style>
     <style name="smallRegularText">
```
bir satır boşluk bırakalım

iamkurtgoz reviewed (commented):
themes içerisinde boşlukları düzenler misin

mustafakamber committed:
feat!: themes.xml and night\themes.xml spaces deleted
feat!: navigationBarColor attributes added

mustafakamber committed:
feat!: themes.xml and night\themes.xml spaces deleted
feat!: navigationBarColor attributes added

mustafakamber committed:
Update README.md

mustafakamber committed:
Update README.md

mustafakamber committed:
feat! : light mode added
